addons/advanced_fatigue/functions/fnc_getWeaponInertia.sqf

```diff
@@ -0,0 +1,36 @@
+#include "script_component.hpp"
+/*
+ * Author: Pterolatypus
+ * Calculates total weapon inertia, accounting for attachments.
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ *
+ * Return Value:
+ * Total inertia <NUMBER>
+ *
+ * Example:
+ * [ACE_player] call ace_advanced_fatigue_fnc_getWeaponInertia
+ *
+ * Public: No
+ */
+params [["_unit", ACE_player, [objNull]]];
+
+private _cache = GVAR(inertiaCache);
+private _weapon = currentWeapon _unit;
+private _accessories = _unit weaponAccessories _weapon;
+private _key = format (["%1$%2+%3+%4+%5", _weapon] + _accessories);
+
+private _inertia = _cache get _key;
+if (isNil "_inertia") then {
+    _inertia = 0;
+    {
+        if (isNumber (configFile >> "CfgWeapons" >> _x >> "inertia")) then {
```

Comment:
`isNumber` returns false for simple expressions, despite `getNumber` returning a NUMBER that can be != 0. This should also do check for `isText`. Consider putting the config path in a variable. Also consider putting `configFile >> "CfgWeapons"` in a higher scope variable.

Reply:
Can a CfgWeapons entry even be missing the inertia property in the first place?

Reply:
You could probably make it so, but then the game likely would use 0. I think you're right and the check is completely unnecessary.

Reply:
A handful of base classes have it undefined so it is possible. I wouldn't be too worried about it if it weren't for nil propagation - I'm a bit nervous about a modder screwing up their inheritance somehow breaking the whole of adv fatigue. This check also catches the case where `_x == ""`.

I don't know if this is allowed to be an expression (it never is in vanilla) but I've changed it to `getNumber` right off the bat and ignore if it's nil, which should cover all bases with fewer commands.